examples/vision/example.jl

```diff
@@ -0,0 +1,93 @@
+using PDDL, Printf
+using SymbolicPlanners, Plinf
+using Gen, GenParticleFilters
+using PDDLViz, GLMakie
+
+include("utils.jl")
+
+# Register PDDL array theory
+println("Registering PDDL array theory")
+PDDL.Arrays.register!()
+
+# Load domain and problem
+println("Loading domain")
+domain = load_domain(joinpath(@__DIR__, "domain.pddl"))
+println("Loading problem")
+problem = load_problem(joinpath(@__DIR__, "problems", "problem-2.pddl"))
```

Comment:
I also tried "problem-1.pddl" but it wasn't working. Is this expected?

```
Registering PDDL array theory
Loading domain
Loading problem
Initializing state
Initial state: GenericState(Set(Compound[onion(onion1), carrot(carrot1)]), Set{Term}(), Dict{Symbol, Any}(:walls => Bool[0 0 0 0; 0 1 1 0; 0 0 0 0; 0 0 0 0], :xloc => Dict{Any, Any}((:onion1,) => 3, (:carrot1,) => 4), :yloc => Dict{Any, Any}((:onion1,) => 3, (:carrot1,) => 4), :viewdistance => 2, :xpos => 1, :ypos => 1))
Constructing goal specification
Compiling domain
ERROR: LoadError: MethodError: Cannot `convert` an object of type Int64 to an object of type BitMatrix
Closest candidates are:
  convert(::Type{T}, ::LinearAlgebra.Factorization) where T<:AbstractArray at /Applications/Julia-1.8.app/Contents/Resources/julia/share/julia/stdlib/v1.8/LinearAlgebra/src/factorization.jl:58
  convert(::Type{<:BitArray}, ::AbstractArray) at bitarray.jl:580
  convert(::Type{T}, ::T) where T<:AbstractArray at abstractarray.jl:16
  ...
Stacktrace:
 [1] setproperty!(x::PDDL.var"##CompiledVisionState#315", f::Symbol, v::Int64)
   @ Base ./Base.jl:39
 [2] set_fluent!(state::PDDL.var"##CompiledVisionState#315", val::Int64, term::Const)
   @ PDDL ~/.julia/packages/PDDL/rIUmS/src/compiler/state.jl:181
 [3] PDDL.var"##CompiledVisionState#315"(state::GenericState)
   @ PDDL ~/.julia/packages/PDDL/rIUmS/src/compiler/state.jl:89
 [4] top-level scope
   @ ~/.julia/packages/PDDL/rIUmS/src/compiler/state.jl:13
 [5] eval
   @ ./boot.jl:368 [inlined]
 [6] eval
   @ ~/.julia/packages/PDDL/rIUmS/src/PDDL.jl:1 [inlined]
 [7] compiled(domain::GenericDomain, state::GenericState)
   @ PDDL ~/.julia/packages/PDDL/rIUmS/src/compiler/compiler.jl:62
 [8] top-level scope
   @ ~/Documents/research/mixi/connectsphere/Plinf.jl/examples/vision/example.jl:27
in expression starting at /Users/nanami/Documents/research/mixi/connectsphere/Plinf.jl/examples/vision/example.jl:27
```

Reply:
You found another bug! Great work :D

It was a problem with a definition in the domain file not matching the problem file. I have removed the viewdistance parameter, so should be okay now

Reply:
I see. Thank you for fixing it so quickly!